src/foam/time/TimeZone.js

```diff
@@ -29,8 +29,23 @@ foam.CLASS({
     {
       name: 'toSummary',
       type: 'String',
-      code: function() { return this.displayName; },
-      javaCode: 'return getDisplayName();'
+      code: function() {
+        var s = this.id;
+        if ( this.displayName && this.id !== this.displayName ) {
+          s += " - ";
+          s += this.displayName;
+        }
+        return s;
+      },
+      javaCode: `
+        StringBuilder sb = new StringBuilder();
+        sb.append(getId());
+        if ( ! foam.util.SafetyUtil.isEmpty(getDisplayName()) ) {
```

Comment:
do we need the equality check for id and displayname too as you have for js?

Reply:
Looks like I started that and didn't finish.  Added.